imports/plugins/core/orders/client/containers/orderDashboardContainer.js

```diff
@@ -194,6 +260,397 @@ class OrderDashboardContainer extends Component {
+      }),
+      type: "warning",
+      showCancelButton: true,
+      showCloseButton: true,
+      allowOutsideClick: false,
+      confirmButtonText: i18next.t("order.approveBulkOrderActionRegression"),
+      cancelButtonText: i18next.t("order.cancelBulkOrderAction")
+    }, (regress) => {
+      if (regress) {
+        // if some of the order(s) want to skip the previous state, display warning alert for skipping states
+        if (falsePreviousStatuses) {
+          this.displayAlert(
+            selectedOrders, status, whichFalseState, falsePreviousStatuses, falseCurrentState, trueCurrentState
+          );
+        } else {
+          // set status of order(s) if this action is confirmed
+          this.shippingStatusUpdateCall(selectedOrders, status);
+        }
+      }
+    });
+  }
+
+  pickedShippingStatus = (selectedOrders, status) => {
+    let isNotPicked = 0;
+    let isPicked = 0;
+    let ordersToRegress = 0;
+
+    selectedOrders.forEach((order) => {
+      // check if the order(s) are in this state already or in the previous state
+      if (order.shipping[0].workflow.status === "new") {
+        isNotPicked++;
+      } else if (order.shipping[0].workflow.status === "coreOrderWorkflow/picked") {
+        isPicked++;
+      } else {
+        // check if the selected order(s) are being regressed back to this state
+        if (order.shipping[0].workflow.workflow.includes("coreOrderWorkflow/picked")) {
+          ordersToRegress++;
+        } else if (!order.shipping[0].workflow.workflow.includes("coreOrderWorkflow/picked") &&
+        (order.shipping[0].workflow.status === "coreOrderWorkflow/packed" ||
+        order.shipping[0].workflow.status === "coreOrderWorkflow/labeled" ||
+        order.shipping[0].workflow.status === "coreOrderWorkflow/shipped")) {
+          ordersToRegress++;
+        }
+      }
+    });
+
+    // display regression alert if order(s) are being regressed
+    if (ordersToRegress) {
+      this.displayRegressionAlert(selectedOrders, ordersToRegress, status);
+    } else {
+      // set status to 'picked' if order(s) are in the previous state
+      if (isNotPicked) {
+        this.shippingStatusUpdateCall(selectedOrders, status);
+
+        // display alert if order(s) are already in this state
+      } else if (!isNotPicked && isPicked) {
+        Alerts.alert({
+          text: i18next.t("order.orderAlreadyInState", {
+            orderText: this.displayOrderText(selectedOrders), status: status
+          })
+        });
+      }
+    }
+  }
+
+  packedShippingStatus = (selectedOrders, status) => {
+    let isNotPicked = 0;
+    let isNotPacked = 0;
+    let isPacked = 0;
+    let ordersToRegress = 0;
+    const whichFalseState = shippingStates.picked;
+
+    selectedOrders.forEach((order) => {
+      // check if the order(s) are in this state already or in one of the previous states
+      if (order.shipping[0].workflow.status === "new") {
```

Comment:
If we're going to leave this in for now, you could clean it up fairly easily with:
`const orderWorkflow = order.shipping[0].workflow.status;` and compare against that rather than using the longer version every time.